src/scenarios/shared/runner.py

```diff
@@ -607,9 +607,22 @@ def run(self):
 
             getLogger().info("Checking device state.")
             cmdline = xharnesscommand() + ['apple', 'state']
-            adb = RunCommand(cmdline, verbose=True)
-            adb.run()
-
+            apple_state = RunCommand(cmdline, verbose=True)
+            apple_state.run()
+
+            # Get the name and version of the device from the output of apple_state above
+            # Example output expected (PERFIOS-01 is the device name, and 17.0.2 is the version):
+            #  Connected Devices:
+            #    PERFIOS-01 00008101-001A09223E08001E    17.0.2        iPhone iOS
+            deviceInfoMatch = re.search(r'Connected Devices:\s+(?P<deviceName>\S+)\s+\S+\s+(?P<deviceVersion>\S+)', apple_state.stdout)
+            if deviceInfoMatch:
+                deviceName = deviceInfoMatch.group('deviceName')
+                deviceVersion = deviceInfoMatch.group('deviceVersion')
+                getLogger().info(f"Device Name: {deviceName}")
+                getLogger().info(f"Device Version: {deviceVersion}")
+            else:
+                raise Exception("Device name or version not found in the output of apple_state command.")
```

Comment:
Should the script continue even when the device details are unknown? I wonder if this is a valid reason to stop the execution.

Reply:
The device name at least needs to be available as it is now required for the launch and kill commands. The version is also used for now to determine the app launch line to look for as they changed slightly sometime between the previous iOS version and iOS 17 or between the XHarness versions.